tests/tests/integration_tests.rs

```diff
@@ -373,3 +373,139 @@ async fn ws_batch_works() {
 	let responses: Vec<String> = client.batch_request(batch).await.unwrap();
 	assert_eq!(responses, vec!["hello".to_string(), "hello".to_string()]);
 }
+
+#[tokio::test]
+async fn http_unsupported_methods_dont_work() {
+	use hyper::{Body, Client, Method, Request};
+
+	let (server_addr, _handle) = http_server().await;
+
+	let http_client = Client::new();
+	let uri = format!("http://{}", server_addr);
+
+	let req_is_client_error = |method| async {
+		let req = Request::builder()
+			.method(method)
+			.uri(&uri)
+			.header("content-type", "application/json")
+			.body(Body::from(r#"{ "jsonrpsee": "2.0", method: "say_hello", "id": 1 }"#))
```

Comment:
I guess the body is irrelevant for this as the we only care of about the HTTP status?

Reply:
That was my thinking; the body was probably tested elsewhere anyway :)